### ISSUE TITLE
Compile error after adding Unity Collections package

### ISSUE THREAD

**Issue by yuchenz27** [ethereum-boilerplate/web3-unity-boilerplate#76]
After adding Unity Collections pakcage, I got this error
```
Library/PackageCache/com.unity.collections@1.1.0/Unity.Collections/xxHash3.cs(560,27): error CS0234: The type or namespace name 'umul128' does not exist in the namespace 'Common' (are you missing an assembly reference?)
Library/PackageCache/com.unity.collections@1.1.0/Unity.Collections/xxHash3.cs(582,27): error CS0234: The type or namespace name 'umul128' does not exist in the namespace 'Common' (are you missing an assembly reference?)
Library/PackageCache/com.unity.collections@1.1.0/Unity.Collections/xxHash3.cs(589,28): error CS0234: The type or namespace name 'umul128' does not exist in the namespace 'Common' (are you missing an assembly reference?)
Library/PackageCache/com.unity.collections@1.1.0/Unity.Collections/xxHash3.cs(825,22): error CS0234: The type or namespace name 'umul128' does not exist in the namespace 'Common' (are you missing an assembly reference?)
```
My platform is iOS.

The second answer of this link explains how to add Unity Collections pakcage

https://forum.unity.com/threads/how-do-i-add-the-collections-package.1138399/

<img width="998" alt="image" src="https://user-images.githubusercontent.com/44870300/156566551-cb6f8bfd-7eea-4ca9-83f1-393f4aee4c55.png">

Does anyone know how to fix this? Thank you.

**Comment by supiket** [comment]
Facing the same issue on Android target.

**Comment by 0xPr0f** [comment]
hey, can you guys drop steps to reproduce ?

**Comment by yuchenz27** [comment]
@0xPr0f sure.
1. Create a new unity project and import moralis boilerplate unitypackage
2. Add com.unity.collections package

I am using Unity2021.2.12 on iOS

**Comment by supiket** [comment]
I solved the issue in the following way:

1. Moved the `com.unity.collections@<your_version>` folder from `<YourProjectRoot>/Library/PackageCache/` directory to `<YourProjectRoot>/Packages/` directory.
2. Opened the file `com.unity.collections@<your_version>/Unity.Collections/xxHash3.cs`.
3. Changed the `Common.umul128(...)` calls to `Unity.Burst.Intrinsics.Common.umul128(...)` calls.

The reason for step 1 is that the `Library` directory is "volatile", so when I edit the script without applying step 1, step 3 was undone with each Unity recompile.

**Comment by xactant** [comment]
@yuchenz27 and @supiket is this a conflict with the Moralis / Unity SDK - re-reading your comments above, it is not apparent that it is. Are you seeing something not reflected in your comments? Thanks.

**Comment by supiket** [comment]
I believe the problem is with `Common.Logging` and/or `Common.Logging.Core` dll files. There is somehow a conflict of the namespace `Common` between these dll files and `Unity.Burst.Intrinsics.Common`.

**Comment by yuchenz27** [comment]
@xactant I think there is a naming conflict between `Unity.Burst.Intrinsics.Common` and these dlls in moralis folder:

![image](https://user-images.githubusercontent.com/44870300/156589921-2125fb97-0769-4739-9ec7-6b3d99933f93.png)

Is is fine to use either of Moralis SDK or Unity Collections package. Those erros appeared if they are used together.

**Comment by supiket** [comment]
Also for reference, [a similar issue](https://github.com/Nethereum/Nethereum/issues/695) was opened for Nethereum.

**Comment by 0xPr0f** [comment]
this is conflict between two packages with the same name

**Comment by yuchenz27** [comment]
@supiket I think your solution is the only quick way to fix that unless Unity upgrades the Collections package or they stop using those `Common.Logging` dlls.

**Comment by xactant** [comment]
Looking through the code, the Common.Logging.DLL is included due to Nethereum. As this is a conflict between two 3RD party packages and not with the Moralis SDK, I am not going to take action on this issue at this time.

Changing status to "wontfix" for now.

**Comment by emandiev** [comment]
The steps suggested by @supiket worked for me as well:

> I solved the issue in the following way:
> 
> 1. Moved the `com.unity.collections@<your_version>` folder from `<YourProjectRoot>/Library/PackageCache/` directory to `<YourProjectRoot>/Packages/` directory.
> 2. Opened the file `com.unity.collections@<your_version>/Unity.Collections/xxHash3.cs`.
> 3. Changed the `Common.umul128(...)` calls to `Unity.Burst.Intrinsics.Common.umul128(...)` calls.

**Comment by KoenRijpstra** [comment]
This problem is fixed version 1.2.0! 🥳

**Comment by JamesW940516** [comment]
Thanks!!! @supiket 
Your solution saved my life.
I am wondering that this issue has been fixed officially or not.
I could not find any proper solution anywhere.
Thanks again!!!
